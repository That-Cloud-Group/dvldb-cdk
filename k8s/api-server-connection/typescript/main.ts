import { Construct } from "constructs";
import { App, Chart, ChartProps } from "cdk8s";

// imported constructs
import { KubeDeployment, KubeService, IntOrString } from "./imports/k8s";

export class ApiServerConnectionContainerChart extends Chart {
  constructor(scope: Construct, id: string, props: ChartProps = {}) {
    super(scope, id, props);

    const label = { app: "dvldb-api-server-connection-container" };

    new KubeService(this, "service", {
      spec: {
        type: "LoadBalancer",
        ports: [{ port: 80, targetPort: IntOrString.fromNumber(8080) }],
        selector: label,
      },
    });

    new KubeDeployment(this, "deployment", {
      spec: {
        replicas: 2,
        selector: {
          matchLabels: label,
        },
        template: {
          metadata: { labels: label },
          spec: {
            containers: [
              {
                name: "nginx",
                image: "nginx:latest",
                ports: [{ containerPort: 8080 }],
                command: ["curl"],
                args: ["http://kubernetes.default.svc.cluster.local/"]
              },
            ],
          },
        },
      },
    });
  }
}

const app = new App();
new ApiServerConnectionContainerChart(app, "dvldb-api-server-connection-container");
app.synth();
