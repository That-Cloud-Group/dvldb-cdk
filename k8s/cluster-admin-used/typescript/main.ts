import * as kplus from "cdk8s-plus-26";
import { Construct } from "constructs";
import { App, Chart, ChartProps } from "cdk8s";

import { KubeClusterRoleBinding } from "./imports/k8s";
// imported constructs

export class ClusterAdminChart extends Chart {
  constructor(scope: Construct, id: string, props: ChartProps = {}) {
    super(scope, id, props);

    const serviceAccount = new kplus.ServiceAccount(
      this,
      "DvlDbAdminServiceAccount"
    );
    new KubeClusterRoleBinding(this, "AdminRoleBinding", {
      metadata: { name: "dvldb-cluster-admin" },
      roleRef: {
        apiGroup: "rbac.authorization.k8s.io",
        kind: "ClusterRole",
        name: "cluster-admin",
      },
      subjects: [
        {
          kind: "ServiceAccount",
          name: serviceAccount.name,
          namespace: "default",
        },
      ],
    });
  }
}

const app = new App();
new ClusterAdminChart(app, "dvldb-cluster-admin-used");
app.synth();
