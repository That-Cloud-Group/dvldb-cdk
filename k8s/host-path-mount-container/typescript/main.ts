import { Construct } from "constructs";
import { App, Chart, ChartProps } from "cdk8s";

// imported constructs
import { KubeDeployment, KubeService, IntOrString } from "./imports/k8s";

export class HostPathMountChart extends Chart {
  constructor(scope: Construct, id: string, props: ChartProps = {}) {
    super(scope, id, props);

    const label = { app: "host-path-mount-container" };

    new KubeService(this, "service", {
      spec: {
        type: "LoadBalancer",
        ports: [{ port: 80, targetPort: IntOrString.fromNumber(8080) }],
        selector: label,
      },
    });

    new KubeDeployment(this, "deployment", {
      spec: {
        replicas: 2,
        selector: {
          matchLabels: label,
        },
        template: {
          metadata: { labels: label },
          spec: {
            containers: [
              {
                name: "nginx",
                image: "nginx:latest",
                volumeMounts: [
                  { mountPath: "/mounted/", name: "host-mounted" },
                ],
                ports: [{ containerPort: 8080 }],
              },
            ],
            volumes: [
              {
                name: "host-mounted",
                hostPath: { path: "/var/log/" },
              },
            ],
          },
        },
      },
    });
  }
}

const app = new App();
new HostPathMountChart(app, "dvldb-host-path-mount-container");
app.synth();
