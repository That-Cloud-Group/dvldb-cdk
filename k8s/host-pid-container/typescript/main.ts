import { Construct } from "constructs";
import { App, Chart, ChartProps } from "cdk8s";

// imported constructs
import { KubeDeployment, KubeService, IntOrString } from "./imports/k8s";

export class HostPidChart extends Chart {
  constructor(scope: Construct, id: string, props: ChartProps = {}) {
    super(scope, id, props);

    const label = { app: "host-pid-container" };

    new KubeService(this, "service", {
      spec: {
        type: "LoadBalancer",
        ports: [{ port: 80, targetPort: IntOrString.fromNumber(8080) }],
        selector: label,
      },
    });

    new KubeDeployment(this, "deployment", {
      spec: {
        replicas: 2,
        selector: {
          matchLabels: label,
        },
        template: {
          metadata: { labels: label },
          spec: {
            hostPid: true,
            containers: [
              {
                name: "nginx",
                image: "nginx:latest",
                ports: [{ containerPort: 8080 }],
              },
            ],
          },
        },
      },
    });
  }
}

const app = new App();
new HostPidChart(app, "dvldb-host-pid-container");
app.synth();
