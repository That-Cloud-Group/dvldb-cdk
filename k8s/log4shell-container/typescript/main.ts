import { Construct } from "constructs";
import { App, Chart, ChartProps } from "cdk8s";

// imported constructs
import { KubeDeployment, KubeService, IntOrString } from "./imports/k8s";

export class Log4shellContainerChart extends Chart {
  constructor(scope: Construct, id: string, props: ChartProps = {}) {
    super(scope, id, props);

    const label = { app: "dvldb-log4shell-container" };

    new KubeService(this, "service", {
      spec: {
        type: "LoadBalancer",
        ports: [{ port: 80, targetPort: IntOrString.fromNumber(8080) }],
        selector: label,
      },
    });

    new KubeDeployment(this, "deployment", {
      spec: {
        replicas: 2,
        selector: {
          matchLabels: label,
        },
        template: {
          metadata: { labels: label },
          spec: {
            containers: [
              {
                name: "tomcat",
                image: "ghcr.io/christophetd/log4shell-vulnerable-app",
                ports: [{ containerPort: 8080 }],
              },
            ],
          },
        },
      },
    });
  }
}

const app = new App();
new Log4shellContainerChart(app, "dvldb-log4shell-container");
app.synth();
