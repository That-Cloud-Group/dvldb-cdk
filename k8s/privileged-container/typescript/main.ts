import { Construct } from "constructs";
import { App, Chart, ChartProps } from "cdk8s";

// imported constructs
import { KubeDeployment, KubeService, IntOrString } from "./imports/k8s";

export class PrivilegedContainerChart extends Chart {
  constructor(scope: Construct, id: string, props: ChartProps = {}) {
    super(scope, id, props);

    const label = { app: "dvldb-privileged-container" };

    new KubeService(this, "service", {
      spec: {
        type: "LoadBalancer",
        ports: [{ port: 80, targetPort: IntOrString.fromNumber(8080) }],
        selector: label,
      },
    });

    new KubeDeployment(this, "deployment", {
      spec: {
        replicas: 2,
        selector: {
          matchLabels: label,
        },
        template: {
          metadata: { labels: label },
          spec: {
            containers: [
              {
                name: "nginx",
                image: "nginx:latest",
                ports: [{ containerPort: 8080 }],
                securityContext: { privileged: true },
              },
            ],
          },
        },
      },
    });
  }
}

const app = new App();
new PrivilegedContainerChart(app, "dvldb-privileged-container");
app.synth();
