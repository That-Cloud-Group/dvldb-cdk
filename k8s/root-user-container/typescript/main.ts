import { Construct } from 'constructs';
import { App, Chart, ChartProps } from 'cdk8s';

// imported constructs
import { KubeDeployment, KubeService, IntOrString } from './imports/k8s';


export class RootContainerChart extends Chart {
  constructor(scope: Construct, id: string, props: ChartProps = { }) {
    super(scope, id, props);

    const label = { app: 'root-user-container' };

    new KubeService(this, 'service', {
      spec: {
        type: 'LoadBalancer',
        ports: [ { port: 80, targetPort: IntOrString.fromNumber(8080) } ],
        selector: label
      }
    });

    new KubeDeployment(this, 'deployment', {
      spec: {
        replicas: 2,
        selector: {
          matchLabels: label
        },
        template: {
          metadata: { labels: label },
          spec: {
            containers: [
              {
                name: 'nginx',
                image: 'nginx:latest',
                ports: [ { containerPort: 8080 } ],
                securityContext: { runAsUser: 0 }
              }
            ]
          }
        }
      }
    });
  }
}

const app = new App();
new RootContainerChart(app, 'dvldb-root-user-container');
app.synth();
