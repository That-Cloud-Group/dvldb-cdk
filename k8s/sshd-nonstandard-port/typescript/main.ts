import { Construct } from "constructs";
import { App, Chart, ChartProps } from "cdk8s";

// imported constructs
import {
  KubeDeployment,
  KubePod,
  KubeService,
  IntOrString,
} from "./imports/k8s";

export class SshdNonstandardPortContainerChart extends Chart {
  constructor(scope: Construct, id: string, props: ChartProps = {}) {
    super(scope, id, props);

    const label = { app: "dvldb-sshd-nonstandard-port-container" };

    new KubeService(this, "service", {
      metadata: {
        name: "nonstandard-sshd-service",
      },

      spec: {
        type: "LoadBalancer",
        ports: [{ port: 8080, targetPort: IntOrString.fromNumber(8080) }],
        selector: label,
      },
    });

    new KubeDeployment(this, "deployment", {
      spec: {
        replicas: 2,
        selector: {
          matchLabels: label,
        },
        template: {
          metadata: { labels: label },
          spec: {
            containers: [
              {
                name: "sshdalpine",
                image: "rjulian/sshdalpine:latest",
                ports: [{ containerPort: 8080 }],
              },
            ],
          },
        },
      },
    });

    new KubePod(this, "sshclient", {
      spec: {
        containers: [
          {
            name: "sshdalpine-client",
            image: "rjulian/sshdalpine:latest",
            command: [
              "sh",
              "-c",
              'ssh -p 8080 -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no root@nonstandard-sshd-service.$(cat /var/run/secrets/kubernetes.io/serviceaccount/namespace).svc.cluster.local "sleep 500"',
            ],
          },
        ],
        initContainers: [
          {
            name: "init-sshd-service",
            image: "busybox:latest",
            command: [
              "sh",
              "-c",
              "until nslookup nonstandard-sshd-service.$(cat /var/run/secrets/kubernetes.io/serviceaccount/namespace).svc.cluster.local; do echo waiting for myservice; sleep 2; done",
            ],
          },
        ],
      },
    });
  }
}

const app = new App();
new SshdNonstandardPortContainerChart(
  app,
  "dvldb-sshd-nonstandard-port-container",
);
app.synth();
