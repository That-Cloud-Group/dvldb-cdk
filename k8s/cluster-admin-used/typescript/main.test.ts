import {ClusterAdminChart} from './main';
import {Testing} from 'cdk8s';

describe('Placeholder', () => {
  test('Empty', () => {
    const app = Testing.app();
    const chart = new ClusterAdminChart(app, 'test-chart');
    const results = Testing.synth(chart)
    
    // Basic infra tests
    expect(results.length).toEqual(2)
    expect(results[0].kind).toEqual("ServiceAccount")
    expect(results[1].kind).toEqual("ClusterRoleBinding")

    // Guarantee insecure
    expect(results[1].roleRef.name).toEqual("cluster-admin")
    expect(results[1].roleRef.kind).toEqual("ClusterRole")

    //  Snapshot tests
    expect(results).toMatchSnapshot();
  });
});
