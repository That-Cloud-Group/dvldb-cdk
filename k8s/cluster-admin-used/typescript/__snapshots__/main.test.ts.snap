// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClusterAdminChart synth Deploy service account and bind cluster role 1`] = `
[
  {
    "apiVersion": "v1",
    "automountServiceAccountToken": false,
    "kind": "ServiceAccount",
    "metadata": {
      "name": "test-chart-dvldbadminserviceaccount-c82124ef",
    },
  },
  {
    "apiVersion": "rbac.authorization.k8s.io/v1",
    "kind": "ClusterRoleBinding",
    "metadata": {
      "name": "dvldb-cluster-admin",
    },
    "roleRef": {
      "apiGroup": "rbac.authorization.k8s.io",
      "kind": "ClusterRole",
      "name": "cluster-admin",
    },
    "subjects": [
      {
        "kind": "ServiceAccount",
        "name": "test-chart-dvldbadminserviceaccount-c82124ef",
        "namespace": "default",
      },
    ],
  },
]
`;
